mixer/doc/deploy-mixer-with-custom-adapter.md

```diff
@@ -0,0 +1,139 @@
+# Deploying Mixer with custom adapters
+
+Congratulations! You've managed to build a complete Mixer adapter and have validated
+that it works via local testing.
+
+Follow these steps to deploy a Mixer with your custom adapter to your service
+mesh for use.
+
+1. Build a docker image of Mixer with your custom adapter.
+
+   Execute the following command:
+
+   ```bash
+   bazel run //mixer/docker:mixer
+   ```
+
+   This should produce output similar to:
+
+   ```
+   INFO: Running command line: bazel-bin/mixer/docker/mixer
+   Loaded image ID: sha256:f2eb3e4f7f98a8ff1a9fa92666e398b102d28da877711f93a7d87e60e692ac8f
+   Tagging f2eb3e4f7f98a8ff1a9fa92666e398b102d28da877711f93a7d87e60e692ac8f as istio/mixer/docker:mixer
+   ```
+
+   Confirm this image is available via:
+
+   ```bash
+   docker images istio/mixer/docker:mixer
+   ```
+
+   The expected output is:
+
+   ```
+   REPOSITORY           TAG                 IMAGE ID            CREATED             SIZE
+   istio/mixer/docker   mixer               f2eb3e4f7f98        47 years ago        169.3 MB
+   ```
+
+1. Tag your new Mixer image appropriately for your image registry of choice.
+
+   If using Google Cloud Registry, use a command similar to:
+   
+   ```bash
+   docker tag istio/mixer/docker:mixer gcr.io/$PROJECT/mixer:$DOCKER_TAG   
+   ```
+   
+   Set `$PROJECT` to your project id and `$DOCKER_TAG` to your desired tag before executing the command.
+   
+1. Push your new Mixer image into the image registry.
+
+   If using Google Cloud Registry, use a command similar to:
+   
+   ```bash
+   gcloud docker -- push gcr.io/$PROJECT/mixer:$DOCKER_TAG
+   ```
+   
+1. Generate the appropriate CRD for your custom adapter.
+
+   The Mixer binary has a utility for generating the Custom Resource Definitions for adapters. Invoke this utility as follows:
+   
+   ```bash
+   bazel run mixer/cmd/server:mixs -- crd adapter
```

Comment:
should we just do `all`, just in case user also has a custom template ? This document can just assume that user is wanting to build a custom mixer with custom adapter with/without custom template.

Reply:
We can certainly expand this to cover custom everything, but I'm worried that the steps might become much more involved or more abstract.

Maybe we can add an appendix ("For custom templates, repeat step X & 7, with the following modifications:...")?